runtime/launcher.m

```diff
@@ -285,23 +275,16 @@
 static int
 push_env (const char *variable, NSString *str_value)
 {
-	const char *value = [str_value UTF8String];
-	size_t len = strlen (value);
 	const char *current;
-	int rv;
+	NSString *value;
 	
 	if ((current = getenv (variable)) && *current) {
-		char *buf = (char *) malloc (len + strlen (current) + 2);
-		memcpy (buf, value, len);
-		buf[len] = ':';
-		strcpy (buf + len + 1, current);
-		rv = setenv (variable, buf, 1);
-		free (buf);
+		value = [NSString stringWithFormat: @"%s:%@", current, str_value];
```

Comment:
I think you reversed the order here:

```suggestion
		value = [NSString stringWithFormat: @"%@:%s", str_value, current];
```

Reply:
I don't think so. It does the same, both what I read (from source) and executed. E.g.

```
int main(int argc, char *argv[])
{
	printf ("NEW %s\n", getenv ("NEW"));
	push_env ("NEW", @"Bonjour");
	printf ("NEW %s\n", getenv ("NEW"));

	printf ("TMPDIR %s\n", getenv ("TMPDIR"));
	push_env ("TMPDIR", @"/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/");
	printf ("TMPDIR %s\n", getenv ("TMPDIR"));
}
```

with existing (old) code
```
$ clang t2.m -framework Foundation
$ ./a.out 
NEW (null)
NEW Bonjour
TMPDIR /var/folders/4p/hr7rrbjj2w5d1ky_yrjw1s380000gn/T/
TMPDIR /var/folders/4p/hr7rrbjj2w5d1ky_yrjw1s380000gn/T/:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/
```

with new ObjC code
```
$ clang t2.m -framework Foundation
$ ./a.out 
NEW (null)
NEW Bonjour
TMPDIR /var/folders/4p/hr7rrbjj2w5d1ky_yrjw1s380000gn/T/
TMPDIR /var/folders/4p/hr7rrbjj2w5d1ky_yrjw1s380000gn/T/:/Library/Frameworks/Xamarin.iOS.framework/Versions/Current/
```

Reply:
Scratch that, source should be `t1.m` in the first one.
Funny thing is that my first version was correct and I inverted it for some reason (I can't see why now).